### ISSUE TITLE
Privacy plugin no longer works with extra_javascript configured

### ISSUE THREAD

**Issue by jonaharagon** [squidfunk/mkdocs-material#5772]
### Context

When I upgrade mkdocs-material and my site dependencies from `9.1.18+insiders.4.37.0` to the current commit, my site can no longer be built.

### Bug description

The build fails when `extra_javascript` is configured in `mkdocs.yml` and the privacy plugin is enabled.

```
INFO    -  DeprecationWarning: pkg_resources is deprecated as an API
             File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/material/plugins/info/plugin.py", line 33, in <module>
               from pkg_resources import get_distribution, working_set
             File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 121, in <module>
               warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: /Users/jonah/GitHub/mkdocs-testing/site
Traceback (most recent call last):
  File "/Users/jonah/GitHub/mkdocs-testing/venv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/mkdocs/__main__.py", line 286, in build_command
    build.build(cfg, dirty=not clean)
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/mkdocs/commands/build.py", line 304, in build
    files = config.plugins.on_files(files, config=config)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/mkdocs/plugins.py", line 533, in on_files
    return self.run_event('files', files, config=config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/mkdocs/plugins.py", line 507, in run_event
    result = method(item, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jonah/GitHub/mkdocs-testing/venv/lib/python3.11/site-packages/material/plugins/privacy/plugin.py", line 110, in on_files
    url = urlparse(path)
          ^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/parse.py", line 394, in urlparse
    url, scheme, _coerce_result = _coerce_args(url, scheme)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/parse.py", line 133, in _coerce_args
    return _decode_args(args) + (_encode_result,)
           ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/parse.py", line 117, in _decode_args
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/parse.py", line 117, in <genexpr>
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
                 ^^^^^^^^
AttributeError: 'ExtraScriptValue' object has no attribute 'decode'
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/12188934/example.zip)

Also, I had to set `plugins.info.archive_stop_on_violation: false`, otherwise it would say `Please update from 9.1.20+insiders.4.37.0 to 9.2.0b2.` even though 9.2.0b2 is not available on the insiders repo. Bonus bug? This reproduction ZIP is built with the latest (https://github.com/squidfunk/mkdocs-material-insiders/commit/d2033e5aee7d006f15b5d872834875f221bc6d23) commit though.

### Steps to reproduce

1. Enable the privacy plugin
2. Add a JS file to `extra_javascript` in mkdocs.yml

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by discdiver** [comment]
Thank you for the report and solution! Affected by the same bug.

**Comment by squidfunk** [comment]
Thanks for reporting. Definitely related to the recent release of MkDocs 1.5.0 which changed internal APIs. We'll upgrade Insiders to MkDocs 1.5.0 and issue a new release later this day. In the meantime, to mitigate, pin MkDocs to 1.4:

```
pip install mkdocs==1.4
```

**Comment by squidfunk** [comment]
> Also, I had to set plugins.info.archive_stop_on_violation: false, otherwise it would say Please update from 9.1.20+insiders.4.37.0 to 9.2.0b2. even though 9.2.0b2 is not available on the insiders repo. Bonus bug? This reproduction ZIP is built with the latest (https://github.com/squidfunk/mkdocs-material-insiders/commit/d2033e5aee7d006f15b5d872834875f221bc6d23) commit though.

Yes, that's still kind of an oddity if versions between Insiders and community edition are not the same. You can open an issue for that as well, so we don't forget it.

**Comment by squidfunk** [comment]
The problem has been resolved in MkDocs 1.5.1 which now allows to use `config.extra_javascript` as before. Additionally, I've resolved another deprecation warning in the social plugin in squidfunk/mkdocs-material-insiders@a71a438a2ca2e073e998d9736db6e120aebcfa9f and updated the dependency to 1.5.1 in squidfunk/mkdocs-material-insiders@64dbfd2a8.

**Comment by squidfunk** [comment]
Released as part of [9.1.21+insiders-4.37.1](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.37.1).

**Comment by jonaharagon** [comment]
I'm unfortunately not seeing this fixed on my end. Attaching a reproduction if you want to double-check I installed the update correctly, but yeah the same reproduction build fails on 9.1.21+insiders.4.37.1.

[example.zip](https://github.com/squidfunk/mkdocs-material/files/12196651/example.zip)

**Comment by squidfunk** [comment]
Did you upgraded your dependencies after pulling the latest release?

**Comment by discdiver** [comment]
Still failing for me, too. And I have 1.5.1 
```
1:57:02 PM: Requirement already satisfied: mkdocs>=1.5.1 in /opt/buildhome/python3.8/lib/python3.8/site-packages (from mkdocs-material==9.1.21+insiders.4.37.1) (1.5.1)
```

**Comment by squidfunk** [comment]
Okay, thanks. Does it reproduce for you with the example provided in the OP?

**Comment by squidfunk** [comment]
Fixed in [`58b7dba42`](https://github.com/squidfunk/mkdocs-material-insiders/commit/58b7dba42d327e6c253c24c959fa5ad8b336757b). and verified with the provided example. Could you both check the `master` of Insiders to verify whether your installations/actions now work again?

**Comment by discdiver** [comment]
Yep, I can confirm that works for me. Thank you!

**Comment by squidfunk** [comment]
Perfect, thanks for confirming! Released as part of [9.1.21+insiders-4.38.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.38.0).

**Comment by jonaharagon** [comment]
Working for me as well 👍
